From 228cc2693ae3125d3af58faed9ca87fe35ba721e Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Thu, 3 Sep 2015 11:51:34 +0200
Subject: [PATCH 0537/4706] drm/i915: Drop port_mst_index parameter from
 pin/eld callback

commit f0675d4a8ed9d3e863ff611561ee0944969a2784 upstream

The port_mst_index parameter was reserved for future use, but
maintainers prefer to add it later when it is actually used.

[Note: this is an update patch to commit [51e1d83cab99: drm/i915: Call
 audio pin/ELD notify function] where I mistakenly applied the older
 version.  Jani and Daniel's review tags were to the latest version,
 so I add them below, too -- tiwai]

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Reviewed-by: Jani Nikula <jani.nikula@intel.com>
Acked-by: Daniel Vetter <daniel@ffwll.ch>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_audio.c |    4 ++--
 include/drm/i915_component.h       |    3 +--
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_audio.c b/drivers/gpu/drm/i915/intel_audio.c
index 969835d..2d52d96 100644
--- a/drivers/gpu/drm/i915/intel_audio.c
+++ b/drivers/gpu/drm/i915/intel_audio.c
@@ -424,7 +424,7 @@ void intel_audio_codec_enable(struct intel_encoder *intel_encoder)
 		dev_priv->display.audio_codec_enable(connector, intel_encoder, mode);
 
 	if (acomp && acomp->audio_ops && acomp->audio_ops->pin_eld_notify)
-		acomp->audio_ops->pin_eld_notify(acomp->audio_ops->audio_ptr, (int) port, 0);
+		acomp->audio_ops->pin_eld_notify(acomp->audio_ops->audio_ptr, (int) port);
 }
 
 /**
@@ -447,7 +447,7 @@ void intel_audio_codec_disable(struct intel_encoder *intel_encoder)
 		dev_priv->display.audio_codec_disable(intel_encoder);
 
 	if (acomp && acomp->audio_ops && acomp->audio_ops->pin_eld_notify)
-		acomp->audio_ops->pin_eld_notify(acomp->audio_ops->audio_ptr, (int) port, 0);
+		acomp->audio_ops->pin_eld_notify(acomp->audio_ops->audio_ptr, (int) port);
 }
 
 /**
diff --git a/include/drm/i915_component.h b/include/drm/i915_component.h
index ab5bde37..b2d56dd 100644
--- a/include/drm/i915_component.h
+++ b/include/drm/i915_component.h
@@ -42,9 +42,8 @@ struct i915_audio_component {
 		 * pin sense and/or ELD information has changed.
 		 * @audio_ptr:		HDA driver object
 		 * @port:		Which port has changed (PORTA / PORTB / PORTC etc)
-		 * @port_mst_index:	Index within that port, for DisplayPort multistreaming
 		 */
-		void (*pin_eld_notify)(void *audio_ptr, int port, int port_mst_index);
+		void (*pin_eld_notify)(void *audio_ptr, int port);
 	} *audio_ops;
 };
 
-- 
1.7.5.4

